.gitlab-ci.yml

```diff
@@ -0,0 +1,92 @@
+default:
+  image:
+    name: gcr.io/kaniko-project/executor:debug
+    entrypoint: [ "" ]
+  tags:
+    - ord1-tenant
+
+stages:
+  - test
+  - generate
+  - build
+  - release
+
+variables:
+  CONTROLLER_FIXED_IMAGE_TAG: ${CI_COMMIT_SHORT_SHA}
+
+.go-cache:
+  variables:
+    GOPATH: $CI_PROJECT_DIR/.go
+  before_script:
+    - mkdir -p .go
+  cache:
+    paths:
+      - .go/pkg/mod/
+
+test:
+  stage: test
+  image: golang:1.18-alpine3.15
+  extends: .go-cache
+  cache:
+    key: bin-${CI_COMMIT_REF_SLUG}
+    paths:
+      - bin/
+  script:
+    - go test ./... -coverprofile cover.out
+
+build:image:
+  stage: build
+  variables:
+    DOCKER_IMG: ${CONTROLLER_IMAGE}
+    REF_IMAGE: $CI_REGISTRY_IMAGE/$DOCKER_IMG:$CI_COMMIT_REF_SLUG
+    FIXED_IMAGE: $CI_REGISTRY_IMAGE/$DOCKER_IMG:$CI_COMMIT_SHORT_SHA
+    DOCKERFILE: ${CI_PROJECT_DIR}/Dockerfile
+  before_script:
+    - export HTTP_PROXY=${CACHE_PROXY}
+    - export NO_PROXY=${CACHE_REGISTRY_HOST}
+    - mkdir -p /kaniko/.docker
+    - echo "{\"auths\":{\"$CI_REGISTRY\":{\"username\":\"$CI_REGISTRY_USER\",\"password\":\"$CI_REGISTRY_PASSWORD\"}}}" > /kaniko/.docker/config.json
+  script: |-
+    /kaniko/executor \
+      $KANIKO_ADDTL_ARGS \
+      --context . \
+      --dockerfile $DOCKERFILE \
+      --destination $REF_IMAGE \
+      --destination $FIXED_IMAGE
+
+tag:release:
+  stage: build
+  image: node:lts-alpine3.15
+  only:
+    refs:
+      - coreweave
+  artifacts:
+    paths:
+      - artifacts.env
+    expire_in: 1 day
+  before_script:
+    - apk --no-cache add git
+  script:
+    - npm install
+    - npx semantic-release
+
+publish:release:
+  stage: release
+  only:
+    refs:
+      - coreweave
+  dependencies:
+    - build:release
+  variables:
+    IMAGE_NAME: {{.}}
+    REF_IMAGE: $CI_REGISTRY_IMAGE/$IMAGE_NAME:$CI_COMMIT_REF_SLUG
+    FIXED_IMAGE: $CI_REGISTRY_IMAGE/$IMAGE_NAME:$CI_COMMIT_SHORT_SHA
+    DOCKERFILE: {{$ciImagesDir}}/{{.}}/Dockerfile
```

Comment:
What's going on here with the {{.}}

Reply:
that was from utility images, copy paste error.